libr/asm/p/asm_x86_nz.c

```diff
@@ -1141,26 +1142,48 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
 			if (argk) {
 				arg++;
 				delta = strchr (arg, '+');
-				//if (!delta) delta = strchr (arg, '-'); // XXX: TODO: handle negative off
 				if (delta) {
 					*delta++ = 0;
 				} else {
 					delta = strchr (arg, '-');
 					if (delta) {
-						ut32 n = -getnum (a, delta + 1);
-						ut8 *N = (ut8 *)&n;
+						int r = getreg (arg);
+						int r2;
+						int r2_is_num = 0;
```

Comment:
yes. I've seen `int` used in other places. Will replace with `bool` if preferable.